src/controller/python/chip/clusters/ClusterObjects.py

```diff
@@ -369,6 +370,19 @@ def _cluster_object(cls) -> ClusterObject:
 
 
 class ClusterEvent(ClusterObject):
+    def __init_subclass__(cls, *args, **kwargs) -> None:
+        """Register a subclass."""
+        super().__init_subclass__(*args, **kwargs)
+        try:
+            if cls.cluster_id not in ALL_EVENTS:
+                ALL_EVENTS[cls.cluster_id] = {}
+            # register this clusterattribute in the ALL_ATTRIBUTES dict for quick lookups
+            ALL_EVENTS[cls.cluster_id][cls.event_id] = cls
+        except NotImplementedError:
```

Comment:
could we have used a `hasattr` to figure out if cluster_id exists or not? Something like:

```python
if hasattr(cls, 'cluster_id') and hasattr(cls, 'event_id'):
  if cls.cluster_id not in ALL_EVENTS:
    ALL_EVENTS[cls.cluster_id] = {}
  ALL_EVENTS[cls.cluster_id][cls.event_id] = cls  
```

Reply:
These are all codegen'd, so they're supposed to be there. I would actually prefer to fail early if the cluster_id isn't there. Maybe something like
if not hasattr(cls, 'cluster_id') or not hasattr(cls, 'event_id'):
  raise BigObviousFailureThatCodeGenIsBroken

Reply:
went a different way and removed the try and fixed the unit tests. These should be there on all subclasses or they're not useful.